File: compiler-group-traslator/src/test/resources/c/onemillionobjecttest.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>
#include "GC.c"
#include "ExceptionHandlingFunctions.c"
VM* vm;
typedef struct Item {
	int val;
	struct Item* next;
}Item;

int main(char* args) {
	vm = newVM();
	struct Item* curr;
	struct Item* head;
	int i;

	head = NULL ;
	for (
	i = 0  ; i < 1000000 ; i++) {
		curr = (Item*)newObject(vm, &curr, sizeof (Item)) ;

		curr->val = i ;

		curr->next = head ;

		head = curr ;
	}

	curr = head ;
	return 0;
}

File: compiler-group-traslator/src/test/resources/onemillionobjecttest.java
class Item {
   public int val;
   public Item next;
}

public class onemillionobjecttest{
	public static void main(String[] args) {
	   Item curr;
	   Item head;
	   int i;

	   head = null;

	   for(i = 0; i < 1000000;i++) {
		  curr = new Item();
		  curr.val = i;
		  curr.next = head;
		  head = curr;
	   }

	   curr = head;
	}
}